### PR TITLE
replace_8bit_linear modules_to_not_convert default value fix

### PR DESCRIPTION
**Pull request by BlackSamorez**
# What does this PR do?

Fixes the default value of `modules_to_not_convert` of `utils.bitsandbytes.replace_8bit_linear`.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@younesbelkada

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks so much for the clean fix!

File: src/transformers/utils/bitsandbytes.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ class `Int8Params` from `bitsandbytes`.
             module._parameters[tensor_name] = new_value
 
 
-def replace_8bit_linear(model, threshold=6.0, modules_to_not_convert="lm_head", current_key_name=None):
+def replace_8bit_linear(model, threshold=6.0, modules_to_not_convert=["lm_head"], current_key_name=None):
```
Comment (amyeroberts) [inline src/transformers/utils/bitsandbytes.py:87]: Please avoid using mutables as default arguments in functions: replace the list with None and set within the function. 

```suggestion
def replace_8bit_linear(model, threshold=6.0, modules_to_not_convert=None, current_key_name=None):
```

and then below on L116: 

```
    modules_to_not_convert = ["lm_head"] if modules_to_not_convert is None else modules_to_not_convert
```

File: src/transformers/utils/bitsandbytes.py [hunk 2]
```diff
@@ -105,8 +105,8 @@ def replace_8bit_linear(model, threshold=6.0, modules_to_not_convert="lm_head",
         threshold (`float`, *optional*, defaults to 6.0):
             `int8_threshold` for outlier detection as described in the formentioned paper. This parameters is set to
             `6.0` as described by the paper.
-        modules_to_not_convert (`str`, *optional*, defaults to `lm_head`):
-            Name of the module to not convert in `Linear8bitLt`. In practice we keep the `lm_head` in full precision
+        modules_to_not_convert (`List[`str`]`, *optional*, defaults to `[`lm_head`]`):
```
Comment (sgugger) [inline src/transformers/utils/bitsandbytes.py:108]:
```suggestion
        modules_to_not_convert (`List[`str`]`, *optional*, defaults to `["lm_head"]`):
```

**Review by sgugger — COMMENTED** [review]
Agree with Amy's comment. Thanks a lot for your PR!

**Review by younesbelkada — APPROVED** [review]
Can confirm the slow tests pass with the new changes, thanks again for the fix!

**Review by amyeroberts — APPROVED** [review]
Thanks for the fix!

**Pull request merged**